packages/blade/src/components/Typography/Text/Text.tsx

```diff
@@ -0,0 +1,97 @@
+import type { ReactElement } from 'react';
+import styled from 'styled-components';
+import type { TextTypes } from '../../../tokens/theme/theme';
+import getPlatform from '../../../utils/getPlatform';
+import type { Theme } from '../../BladeProvider';
+import BaseText from '../BaseText';
+import type { BaseTextProps } from '../BaseText/BaseText';
+
+type TextCommonProps = {
+  type: TextTypes;
+  truncateAfterLines?: number;
+  children?: React.ReactNode;
+};
+
+type TextBodyVariant = TextCommonProps & {
+  variant: 'body';
+  weight: keyof Theme['typography']['fonts']['weight'];
+};
+
+type TextCaptionVariant = TextCommonProps & {
+  variant: 'caption';
+  weight: keyof Pick<Theme['typography']['fonts']['weight'], 'regular'>;
+};
+
+export type TextProps<T> = T extends {
+  variant: infer Variant;
+}
+  ? Variant extends 'caption'
+    ? TextCaptionVariant
+    : Variant extends 'body'
+    ? TextBodyVariant
+    : T
+  : T;
+
+type TextForwardedAs = {
+  forwardedAs?: BaseTextProps['as'];
+};
+
+const getProps = <T extends { variant: 'body' | 'caption' }>({
+  variant,
+  type,
+  weight,
+}: Pick<TextProps<T>, 'type' | 'variant' | 'weight'>): BaseTextProps & TextForwardedAs => {
+  const props: BaseTextProps & TextForwardedAs = {
+    color: `surface.text.${type}.lowContrast`,
```

Comment:
Getting an eslint error here. Are you not getting this?

<img width="1218" alt="Screenshot 2022-05-04 at 5 08 35 PM" src="https://user-images.githubusercontent.com/24487274/166674133-d1d6f6b7-d6a9-4f4d-95c6-a18f06f2797f.png">

Reply:
nupe, even the validate check passed 🤔